File: 101/scala/Factorial.scala
import scala.annotation.tailrec

/**
 * Find the factorial (n!) of a number (n), in this case 5.
 */
object Factorial {

  def main(args: Array[String]) = println(factorial(5))

  /**
   * Recursively compute the factorial of a number.
   *
   * @param number the number to compute for.
   * @return the computed factorial of specified number.
   */
  def factorial(number: Long) = {
    @tailrec
    def factorialRecursion(number: Long, accumulator: Long = 1): Long =
      if(number > 1) factorialRecursion(number - 1, number * accumulator) else accumulator

    factorialRecursion(number)
  }
}

File: 101/js/factorial.js
/**
 * Calculate the factorial of a number.
 *
 * @param num the number to calculate the factorial of.
 * @returns {number} the factorial of specified number.
 */
const factorial = (num)  =>{
  if(num == 1) {
    return 1;
  } else {
    return num * factorial(num - 1);
  }
};

module.exports = factorial;